packages/circuit-ui/components/Modal/Modal.stories.tsx

```diff
@@ -13,170 +13,160 @@
  * limitations under the License.
  */
 
-import { MouseEvent, KeyboardEvent } from 'react';
-import styled from '@emotion/styled';
+/* eslint-disable react/display-name */
+import { Fragment } from 'react';
 import { css } from '@emotion/core';
-import { action } from '@storybook/addon-actions';
+import { Theme } from '@sumup/design-tokens';
 
+import { Stack } from '../../../../.storybook/components';
 import Button from '../Button';
-import ButtonGroup from '../ButtonGroup';
 import Body from '../Body';
+import Image from '../Image';
+import { ModalProvider } from '../ModalContext';
+import { spacing } from '../../styles/style-mixins';
 
 import docs from './Modal.docs.mdx';
-import { ModalWrapper, ModalHeader, ModalFooter } from './components';
-import { ModalConsumer, ModalProvider } from './ModalContext';
 import { Modal, ModalProps } from './Modal';
+import { useModal } from './useModal';
 
 export default {
   title: 'Components/Modal',
   component: Modal,
+  subcomponents: { ModalProvider },
   parameters: {
     docs: { page: docs },
   },
 };
 
-/* eslint-disable react/display-name, react/prop-types */
-
-const PageWithModal = (modal: ModalProps) => (
-  <ModalProvider>
-    <ModalConsumer>
-      {({ setModal }) => (
-        <Button
-          type="button"
-          onClick={() => {
-            setModal(modal);
-          }}
-        >
-          Open modal
-        </Button>
-      )}
-    </ModalConsumer>
-  </ModalProvider>
-);
-
-const defaultModal = {
-  children: () => <ModalWrapper>Hello World!</ModalWrapper>,
-  onClose: () => {},
+export const Base = (modal: ModalProps): JSX.Element => {
+  const ComponentWithModal = () => {
+    const { setModal } = useModal();
+
+    return (
+      <Button type="button" onClick={() => setModal(modal)}>
+        Open modal
+      </Button>
+    );
+  };
+  return (
+    <ModalProvider>
+      <ComponentWithModal />
+    </ModalProvider>
+  );
+};
+
+Base.args = {
```

Comment:
Nitpick, I'd add a default `closeButtonLabel` here so that someone trying out the component on the Storybook sees the effect of toggling `preventClose` directly. Otherwise they'd need to toggle it _and_ add a label.

Reply:
Good idea!

Reply:
Done in 00e30a1.